sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionAnsiErrorsSuite.scala

```diff
@@ -403,4 +403,16 @@ class QueryExecutionAnsiErrorsSuite extends QueryTest
       parameters = Map("timeZone" -> "-100")
     )
   }
+
+  test("SPARK-50290: Add a flag to disable DataFrame context") {
+    withSQLConf(SQLConf.DATA_FRAME_QUERY_CONTEXT_ENABLED.key -> "false") {
+      val df = spark.range(0, 1).select(lit(1) / col("id"))
+      checkError(
+        exception = intercept[SparkArithmeticException](df.collect()),
+        condition = "DIVIDE_BY_ZERO",
+        parameters = Map("config" -> ansiConf),
+        context = ExpectedContext("", -1, -1)
```

Comment:
hmm, why do we still provide this empty context?

Reply:
The context is always passed to `divideByZeroError`:
![image](https://github.com/user-attachments/assets/7018b720-96e2-4cb7-9040-0e8b1c27aa24)

Reply:
I see, thanks.